Automattic/jetpack
# Update allow list settings access

Fixes #
- Updates the allow list logic so that the settings are accessible regardless of the state of the related security features.
- Collapses Allow and Block lists in the Jetpack security settings when disabled or unpopulated.
- General Protect UI adjustments as per design.

### Other information:


## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/1359

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Checkout this branch for Protect and Jetpack on a fresh JN instance using the beta tester
* Run through the various scenarios outlined in the screenshots below
* Ensure the proposed fixes/updates apply
* Ensure that no regressions are introduced

## Screenshots

Jetpack block list enabled:
![Screen Shot on 2024-08-14 at 15-01-36](https://github.com/user-attachments/assets/24cbc692-f25c-41e1-882b-5a75c139bfd7)

Jetpack block list disabled:
![Screen Shot on 2024-08-14 at 14-58-21](https://github.com/user-attachments/assets/39291c27-f59b-427b-ae8e-6f5c78d6a215)

Jetpack block list disabled (list populated):
![Screen Shot on 2024-08-14 at 15-02-14](https://github.com/user-attachments/assets/bd41eff3-41c1-44a2-8d1c-1d332d29437f)

Jetpack allow list enabled:
![Screen Shot on 2024-08-14 at 15-35-46](https://github.com/user-attachments/assets/0be3c02e-aaf8-4a4e-8fcf-1564b10e2b37)

Jetpack allow list disabled:
![Screen Shot on 2024-08-14 at 15-36-17](https://github.com/user-attachments/assets/73b817c8-bdc2-4e00-98b9-fc69faa45411)

Jetpack allow list disabled (list populated):
![Screen Shot on 2024-08-14 at 15-36-56](https://github.com/user-attachments/assets/5b51ce11-6eb3-4e25-b30b-9cbccbbedc15)

Jetpack Allow list settings always accessible (when all sec features inactive):
   - Supported:
![Screen Shot on 2024-08-14 at 15-48-35](https://github.com/user-attachments/assets/99fea3ac-dded-4b4d-8c1f-62195743909a)
   - Unsupported:
![Screen Shot on 2024-08-14 at 15-49-42](https://github.com/user-attachments/assets/d1b6901a-a3be-4395-94d5-e97034c51917)

Protect allow/block list enabled:
![Screen Shot on 2024-08-14 at 15-53-28](https://github.com/user-attachments/assets/b33787b6-828e-48da-873d-bf6737a54c7d)

Protect allow/block list disabled:
![Screen Shot on 2024-08-14 at 15-54-22](https://github.com/user-attachments/assets/94666395-46f9-46c6-9d70-b32ecc1b38cc)

Protect allow/block list disabled (list populated):
![Screen Shot on 2024-08-14 at 15-54-58](https://github.com/user-attachments/assets/92d64bf5-925b-44cb-b6a8-b35ee308a59f)

Protect Allow list settings always accessible (when all sec features inactive):
   - Supported:
![Screen Shot on 2024-08-14 at 15-51-47](https://github.com/user-attachments/assets/53ebac3c-13ab-4276-bb89-1c5e2b254645)
   - Unsupported:
![Screen Shot on 2024-08-14 at 15-50-55](https://github.com/user-attachments/assets/e0937827-ae15-4cf1-a6d5-722cae03daa1)

General adjustment to Protect Firewall UI as per @ilonagl's proposal:
![Screen Shot on 2024-08-14 at 16-46-55](https://github.com/user-attachments/assets/f4ab6552-6c56-4dfb-a65a-4d50b35127e4)
